src/replicated_log.rs

```diff
@@ -43,10 +47,29 @@ impl<IO: Io> ReplicatedLog<IO> {
     /// また、以前のノードを再起動したい場合でも、もし永続ストレージが壊れている等の理由で、
     /// 前回の状態を正確に復元できないのであれば、
     /// ノード名を変更して、新規ノード追加扱いにした方が安全である.
-    pub fn new(node_id: NodeId, members: ClusterMembers, io: IO) -> Self {
+    #[allow(clippy::new_ret_no_self)]
+    pub fn new(
+        component: ComponentId,
```

Comment:
I recommend to take `MetricBuilder` instead of `ComponentId` as the argument because the former is more flexible.

Reply:
Ok.